lib/service/http.rb

```diff
@@ -140,5 +140,32 @@ def shorten_url(url)
     rescue TimeoutError
       url
     end
+
```

Comment:
These methods were pulled up into this module for reuse across components; this should improve consistency of error messaging and handling when response status can help identify the source of problems.